builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -354,10 +354,12 @@ func resourceArmVirtualMachine() *schema.Resource {
 						"provision_vm_agent": {
 							Type:     schema.TypeBool,
 							Optional: true,
+							Default:  false,
```

Comment:
If we are changing these to have a default, we need to include a schema migration IMO